src/datasources.js

```diff
@@ -1,3 +1,45 @@
+/**
+ * Asynchronously loads data sources from vellum.opts().core.dataSources
+ * Currently only supports fixtures
+ *
+ * Format in opts:
+ * dataSources: [
+ *     {
+ *          key: string : data source name (fixture, case, products...)
+ *          endpoint: function or string (URL)
+ *     }
+ * ]
+ *
+ * The endpoint function (or URL) should return the following format:
+ * [
+ *      {
+ *          sourceUri: string (used in the instance definition)
+ *          defaultId: string (used in instance definition)
+ *          intialQuery: string (used in nodeset)
+ *          name: string (text in the dropdown)
+ *          structure: nested dictionary of attributes
+ *          {
+ *              attribute1: {
+ *                  structure: {
+ *                      inner-attribute1: {
+ *                          no_option: boolean (does not show up in the dropdown)
+ *                      }
+ *                  }
+ *              }
+ *          }
+ *      }
+ * ]
```

Comment:
Yeah. I can't really think of a great reason to keep it at the moment. My original thought was that we are using a dropdown and if we added cases/locations to that dropdown there would be a lot of noise.

However, before we add those two we should probably change the dropdown to a combobox so that the noise wouldn't matter as much.

I think I just tried to over engineer it.